# HG changeset patch
# User Rik <rik@octave.org>
# Date 1541200309 25200
#      Fri Nov 02 16:11:49 2018 -0700
# Node ID 8aaf5f597afc72a8b98cc820d9b87a57d4c6d5e0
# Parent  2e97124a71dafee3e5b35757593ae17ca1f7b98b
__uiobject_split_args__.m: Remove input validation for even number of PROP/VALUE pairs (bug #44672).

* __uiobject_split_args__.m: Remove input validation for even number of PROP/VALUE pairs.
The C++ code in set() already takes care of the validation.

diff --git a/scripts/gui/private/__uiobject_split_args__.m b/scripts/gui/private/__uiobject_split_args__.m
--- a/scripts/gui/private/__uiobject_split_args__.m
+++ b/scripts/gui/private/__uiobject_split_args__.m
@@ -35,21 +35,16 @@ function [parent, args] = __uiobject_spl
       offset = 2;
     elseif (! ischar (in_args{1}) && ! isstruct (in_args{1}))
       error ("%s: invalid parent handle.", who);
     endif
 
     args = in_args(offset:end);
   endif
 
-  numpairs = numel (args) - sum (cellfun ("isclass", args, "struct"));
-  if (rem (numpairs, 2) == 1)
-    error ("%s: PROPERTY/VALUE arguments must occur in pairs", who);
-  endif
-
   if (! isempty (args))
     i = find (strcmpi (args, "parent"), 1, "first");
     if (! isempty (i) && numel (args) > i)
       parent = args{i+1};
       if (! ishghandle (parent))
         error ("%s: invalid parent handle.", who);
       endif
       args(i:i+1) = [];
